pyccel/codegen/utilities.py

```diff
@@ -70,7 +70,13 @@ def construct_flags(compiler,
 
     if accelerator is not None:
         if accelerator == "openmp":
-            flags += " -fopenmp"
+            if sys.platform == "darwin":
+                if compiler == "gcc":
+                    flags += " -Xpreprocessor -fopenmp -lomp"
+                else:
+                    flags += " -fopenmp"
+            else:
+                flags += " -fopenmp"
```

Comment:
the `-fopenmp` must e following the `-Xpreprocessor` otherwise it fail

Reply:
Then
```
if sys.platform == "darwin" and compiler == "gcc":
    flags += " -Xpreprocessor -lomp"
flags += " -fopenmp"
```
?

Reply:
the same error :
clang: error: unsupported option '-fopenmp'

Reply:
> the same error :
> `clang: error: unsupported option '-fopenmp'`

How is that possible. Is this not the same thing as what is done now?

Reply:
i dont think its the same think.
see here : 
https://iscinumpy.gitlab.io/post/omp-on-high-sierra/

Reply:
There isn't an `else` in my code. The only difference is the placement of the `-lomp` but that will move anyway when it is correctly saved to `libs` rather than `flags`